digid_eherkenning/models/eherkenning_metadata_config.py

```diff
@@ -0,0 +1,201 @@
+            "can be a string (the name of the attribute) or an object with keys 'name' "
+            "and 'required', where 'name' is a string and 'required' a boolean'."
+        ),
+    )
+    eidas_requested_attributes = models.JSONField(
+        _("requested attributes"),
+        default=list,
+        help_text=_(
+            "A list of additional requested attributes. A single requested attribute "
+            "can be a string (the name of the attribute) or an object with keys 'name' "
+            "and 'required', where 'name' is a string and 'required' a boolean'."
+        ),
+    )
+    oin = models.CharField(
+        _("Oin"),
+        help_text=_("The OIN of the company providing the service."),
+        max_length=100,
+    )
+    no_eidas = models.BooleanField(
+        _("No eidas"),
+        blank=True,
+        default=False,
+        help_text=_(
+            "If True, then the service catalogue will contain only the eHerkenning service."
+        ),
+    )
+    privacy_policy = models.URLField(
+        _("Privacy policy"),
+        help_text=_(
+            "The URL where the privacy policy from the organisation providing the service can be found."
+        ),
+        max_length=100,
+    )
+    makelaar_id = models.CharField(
+        _("Makelaar ID"),
+        help_text=_("OIN of the broker used to set up eHerkenning/eIDAS."),
+        max_length=100,
+    )
+    artifact_resolve_content_type = models.CharField(
+        _("resolve artifact binding content type"),
+        default="application/soap+xml",
+        max_length=100,
+    )
+    service_language = models.CharField(
+        _("service language"),
+        max_length=2,
+        default="nl",
+        help_text=_("Metadata for eHerkenning/eidas will contain this language key"),
+    )
+
+    class Meta:
+        verbose_name = _("Eherkenning metadata configuration")
+
+    def as_dict(self) -> dict:
+        """
+        Emit the configuration as a dictionary compatible with the old settings format.
+        """
+        organization = None
+
+        if (
+            not self.certificate
+            or not self.certificate.private_key
+            or not self.certificate.public_certificate
+        ):
+            raise ImproperlyConfigured(
+                "No (valid) certificate configured. The configuration needs a "
+                "certificate with private key and public certificate."
+            )
+
+        if self.organization_url and self.organization_name:
+            organization = {
+                self.service_language: {
+                    "name": self.organization_name,
+                    "displayname": self.organization_name,
+                    "url": self.organization_url,
+                }
+            }
+
+        # at least the EH service
+        services = [
+            {
+                # FIXME - the UUID should probably not change every time!!
```

Comment:
TODO -> discuss with Joeri & Silvia

Reply:
so we'll need a way to either read this from an older file, or provide it as a CLI/config argument, right?

Reply:
Yes! In this PR (which was closed) I had added the arguments https://github.com/maykinmedia/django-digid-eherkenning/pull/2/files. But then the main feature of the PR was not needed anymore and I didn't cherry-pick the addition of the parameters :S